core/shared/src/test/scala/fs2/StreamCombinatorsSuite.scala

```diff
@@ -834,6 +835,77 @@ class StreamCombinatorsSuite extends Fs2Suite {
         )
         .assertEquals(0.millis)
     }
+
+    test("upstream failures are propagated downstream") {
+      TestControl.executeEmbed {
+        case object SevenNotAllowed extends NoStackTrace
+
+        val source = Stream
+          .iterate(0)(_ + 1)
+          .covary[IO]
+          .evalTap(n => IO.raiseError(SevenNotAllowed).whenA(n == 7))
+
+        val downstream = source.groupWithin(100, 2.seconds)
+
+        downstream.intercept[SevenNotAllowed.type]
```

Comment:
Should we make an assertion here about what the downstream has / has not received before the error?